src/NovelRT/Graphics/FontSet.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#include <NovelRT.h>
+
+namespace NovelRT::Graphics {
+  FontSet::FontSet(std::weak_ptr<RenderingService> renderer, Atom id) noexcept :
+    _renderer(renderer),
+    _id(id),
+    _fontFile("") {
+
+  }
+
+  void FontSet::loadFontAsTextureSet(const std::string& file, float fontSize) {
+    if (!_fontFile.empty()) {
+      _logger.logErrorLine("This FontSet has already been initialised with data. Please make a new FontSet!");
+      throw std::runtime_error("Unable to continue! Cannot overwrite FontSet!");
+      }
+
+      if (_fontCharacters.size() != 0) _fontCharacters.clear();
```

Comment:
This and the following line are very hard to read.

I think its because the if doesn't have braces and the "contents" are on the same line. So my first assumption is that `FT_Library freeTypeLoader` is improperly indented